### ISSUE TITLE
Own Plugins?

### ISSUE THREAD

**Issue by tobias-walle** [BoltAI/BoltAI#55]
I purchased the premium license. On the [buy page](https://boltai.com/buy) it says that this should allow you to create your own plugins.

<img width="478" alt="image" src="https://github.com/BoltAI/BoltAI/assets/9933601/8196393b-8761-483b-b47b-5bd7cc0b8291">

Is this feature available yet? If yes, is there some documentation on how to use it?

**Comment by longseespace** [comment]
Hey Tobias. Sorry for the confusion. It's not available yet. I'm working hard on releasing this feature as it requires quite a lot of consideration regarding security & privacy.

ETA: will be release some time next month (April).

Stay tuned!

**Comment by tobias-walle** [comment]
Understandable 🙂 Looking forward to the official release!

**Comment by juandrei** [comment]
All function-calling features will make BoltAI stand out from the crowd, for more advanced users. In the spirit of making BoltAI different, it would be great that your release contains useful templates of "future-oriented" cases. Cases I'd find useful are those involving calling API services, Google Map queries, or solving math with Wolfram Alpha. 

Related, since BoltAI already provides access to many different LLMs, it would be great to have a "graphical agentic framework", or better yet, a GUI for creating cross-LLM workflows. BoltAI 2.0?  Thanks for your kind consideration !

**Comment by longseespace** [comment]
@juandrei yes, there is a thread on building prompt chaining and LLM workflow on Canny. It requires a lot of engineering time though. But yes, it's a powerful feature and I will think more about it once the plugin system is finished :D

**Comment by longseespace** [comment]
Sorry for the slow progress on this one. I, again, underestimated the effort needed 😅

Building first-party plugin is simple. I can ship fast by skipping steps, or freely access sensitive data such as API keys.

But for a third-party plugin, this requires stricter access controls, and in the standard way (relatively stable schema etc)

I've made some progress but not enough to release yet.

So the next version I will focus on some new plugins first (first-party) and will continue to support custom plugin in the version after that.

Thanks for your patience 🙏
